### ISSUE TITLE
(rds): instanceUpdateBehaviour is broken with `writers`/`readers` configuration

### ISSUE THREAD

**Issue by blimmer** [aws/aws-cdk#27694]
### Describe the bug

According to [the documentation](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_rds-readme.html#updating-the-database-instances-in-a-cluster), I should be able to use the `instanceUpdateBehaviour` setting with the new `writers`/`readers` (e.g., not `instanceProps`) configuration:

![Screenshot 2023-10-26 at 08 41 00](https://github.com/aws/aws-cdk/assets/630449/f38c1c53-c620-4632-bbd3-a2f58c15abee)

However, my `cdk diff` shows that even though the setting is still set, the dependency is removed (effectively undoing the `ROLLING` update setting):

```
Stack Database
Resources
[~] AWS::RDS::DBInstance db/Instance2 dbInstance2648805E1
 └─ [-] DependsOn
     └─ ["dbInstance1FD14D62F"]
```

Looking at the source code, `instanceUpdateBehaviour` is only paid attention to in the `legacyCreateInstances` method, called when using `instanceProps` https://github.com/aws/aws-cdk/blob/22a3234d7accb9ec78b87bfc8ca5fe6537b769c6/packages/aws-cdk-lib/aws-rds/lib/cluster.ts#L1233 https://github.com/aws/aws-cdk/blob/22a3234d7accb9ec78b87bfc8ca5fe6537b769c6/packages/aws-cdk-lib/aws-rds/lib/cluster.ts#L1330-L1335

When using `writers`/`readers`, this parameter is ignored https://github.com/aws/aws-cdk/blob/22a3234d7accb9ec78b87bfc8ca5fe6537b769c6/packages/aws-cdk-lib/aws-rds/lib/cluster.ts#L634

### Expected Behavior

I expected the `instanceUpdateBehaviour` to work still as I transitioned from `instanceProps` to the new API. The property is not marked as `@deprecated` and the documentation explicitly shows an example of it working with `writers`/`readers`.

### Current Behavior

The CloudFormation dependency is marked for removal in the `cdk diff` when I transition from `instanceProps` to the new `writers`/`readers` API:

```
Stack Database
Resources
[~] AWS::RDS::DBInstance db/Instance2 dbInstance2648805E1
 └─ [-] DependsOn
     └─ ["dbInstance1FD14D62F"]
```

### Reproduction Steps

To show this issue, you can `cdk synth` with the `instanceBehaviour` specified and not specified. The CFN produced is exactly the same.

Given,

1. Create this stack in a new CDK repo:
	```ts
	import * as cdk from 'aws-cdk-lib';
	import * as rds from 'aws-cdk-lib/aws-rds';
	import * as ec2 from 'aws-cdk-lib/aws-ec2';
	import { Construct } from 'constructs';
	
	export class DatabaseStack extends cdk.Stack {
	  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
	    super(scope, id, props);
	
	    const vpc = new ec2.Vpc(this, 'VPC');
	    new rds.DatabaseCluster(this, 'Database', {
	      engine: rds.DatabaseClusterEngine.auroraMysql({ version: rds.AuroraMysqlEngineVersion.VER_3_01_0 }),
	      writer: rds.ClusterInstance.provisioned('Instance1' ),
	      readers: [rds.ClusterInstance.provisioned('Instance2')],
	      instanceUpdateBehaviour: rds.InstanceUpdateBehaviour.ROLLING,
	      vpc,
	    });
	  }
	}
	```
2. Produce a template and save it off

	```shell
    > yarn cdk synth > instance-behaviour-set.yml
    ```
3. Update the stack to not set `instanceUpdateBehaviour`:
	```ts
	import * as cdk from 'aws-cdk-lib';
	import * as rds from 'aws-cdk-lib/aws-rds';
	import * as ec2 from 'aws-cdk-lib/aws-ec2';
	import { Construct } from 'constructs';
	
	export class DatabaseStack extends cdk.Stack {
	  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
	    super(scope, id, props);
	
	    const vpc = new ec2.Vpc(this, 'VPC');
	    new rds.DatabaseCluster(this, 'Database', {
	      engine: rds.DatabaseClusterEngine.auroraMysql({ version: rds.AuroraMysqlEngineVersion.VER_3_01_0 }),
	      writer: rds.ClusterInstance.provisioned('Instance1' ),
	      readers: [rds.ClusterInstance.provisioned('Instance2')],
	      // instanceUpdateBehaviour: rds.InstanceUpdateBehaviour.ROLLING, <- CHANGED
	      vpc,
	    });
	  }
	}
	```
4. Produce another template

	```shell
    > yarn cdk synth > instance-behaviour-not-set.yml
    ```
5. Diff the two templates:

    ```shell
	> diff instance-behaviour-set.yml instance-behaviour-not-set.yml
	571c571
	< Done in 2.75s.
	---
	> Done in 2.28s
    ```

Note that the `instanceUpdateBehaviour` property has no effect on the resulting template.

### Possible Solution

1. Add the instance update behavior logic to the new `_createInstances` method used by the `writers`/`readers` API.
2. If `instanceUpdateBehaviour` does not make sense with the new API, mark it as deprecated and remove the example (linked above) from the documentation.

### CDK CLI Version

2.103.0 (build d0d7547)

### Node.js Version

18.18.2

### OS

MacOS

### Language

TypeScript

**Comment by pahud** [comment]
I believe `instanceUpdateBehaviour` is  just temporary for transitioning to serverless v2 support and should be marked as deprecated in future versions.

**Comment by blimmer** [comment]
@pahud - I'm not sure that it's only useful in moving to Serverless V2 support. I think that when you're using standard provisioned instances, it's useful to support the `ROLLING` update behavior. This ensures that if instances need to be replaced, they don't all go down at once (`BULK` update).

Based on @spanierm42's original PR and the original issue https://github.com/aws/aws-cdk/issues/10595 filed by @hixi-hyi , this implementation didn't really even consider Serverless v2 when created.

Based on that, I think this issue might be best considered a regression with the new `writer`/`readers` API? What do you all think?

**Comment by danstoner** [comment]
> it's useful to support the ROLLING update behavior. This ensures that if instances need to be replaced, they don't all go down at once (BULK update).

I agree with this!  There many cases where clusters will not be Serverless and still need rolling behavior to avoid noticeable service interruptions.
